packages/delegation-toolkit/src/actions/infuraBundlerClient.ts

```diff
@@ -0,0 +1,114 @@
+import type { Transport, Chain, Hex } from 'viem';
+import {
+  createBundlerClient,
+  type BundlerClientConfig,
+  type SmartAccount,
+} from 'viem/account-abstraction';
+
+/**
+ * Gas price tiers returned by pimlico_getUserOperationGasPrice
+ */
+export type GasPriceTier = {
+  /** Maximum fee per gas in hex format */
+  maxFeePerGas: Hex;
+  /** Maximum priority fee per gas in hex format */
+  maxPriorityFeePerGas: Hex;
+};
+
+/**
+ * Response from pimlico_getUserOperationGasPrice RPC method
+ */
+export type UserOperationGasPriceResponse = {
+  /** Slow gas price tier */
+  slow: GasPriceTier;
+  /** Standard gas price tier */
+  standard: GasPriceTier;
+  /** Fast gas price tier */
+  fast: GasPriceTier;
+};
+
+/**
+ * Infura bundler actions for extending bundler clients.
+ *
+ * @returns A function that takes a bundler client and returns the client extension with Infura bundler actions.
+ */
+const infuraBundlerActions = () => (client: any) => ({
+  /**
+   * Get user operation gas prices from Infura bundler.
+   * Calls the pimlico_getUserOperationGasPrice RPC method.
+   *
+   * @returns Promise resolving to gas price tiers (slow, standard, fast).
+   * @example
+   * ```typescript
+   * const gasPrices = await bundlerClient.getUserOperationGasPrice();
+   * console.log(gasPrices.standard.maxFeePerGas);
+   * ```
+   */
+  async getUserOperationGasPrice(): Promise<UserOperationGasPriceResponse> {
+    const response = await client.request({
+      method: 'pimlico_getUserOperationGasPrice' as any,
+      params: [] as any,
+    });
+
+    return response as unknown as UserOperationGasPriceResponse;
+  },
+});
+
+/**
+ * Type for bundler client extended with Infura bundler actions.
+ */
+export type InfuraBundlerClient<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TAccount extends SmartAccount | undefined = SmartAccount | undefined,
+> = ReturnType<typeof createBundlerClient<TTransport, TChain, TAccount>> &
```

Comment:
Viem exports a concrete BundlerClient type that you can use here to simplify this statement.
```suggestion
> = BundlerClient<TTransport, TChain, TAccount> &
```

If the developer attempts to follow the typing, and just ends up at a definition that's deriving the type from other functions, it could be confusing. It might make the type discovery easier for consumers, if we explicitly defined the expected functions that are added to the client:

```
BundlerClient<TTransport, TChain, TAccount> & {
  getUserOperationGasPrice: () => Promise<UserOperationGasPriceResponse>;
};
```

Reply:
sure